placeSpecificContent/hq/courtyard/controlPanel/controlPanel_server.js

```diff
@@ -0,0 +1,406 @@
+         */
+        pressButton: function(thisID, params) {
+            if (!ready) {
+                return;
+            }
+
+            courtyardInPlace = false;
+            roundtableInPlace = false;
+            courtyardVelocity = 0;
+            roundtableVelocity = 0;
+
+            var buttonID = params[0];
+            
+            var targetCourtyardPosition = BUTTON_TO_ENTITY_POSITIONING_MAP[buttonID].courtyardPosition;
+            var targetRoundtablePosition = BUTTON_TO_ENTITY_POSITIONING_MAP[buttonID].roundtablePosition;
+
+            _this.setUpMovement(roundtable, targetRoundtablePosition);
+            _this.setUpMovement(courtyard, targetCourtyardPosition);
+
+            Object.keys(BUTTON_TO_ENTITY_POSITIONING_MAP).forEach(function(button) {
+                if (button === buttonID) {
+                    Entities.editEntity(button, {
+                        backgroundColor: BUTTON_TRANSITION_COLOR,
+                        textColor: BUTTON_INACTIVE_COLOR
+                    });
+                } else {
+                    Entities.editEntity(button, {
+                        backgroundColor: BUTTON_INACTIVE_COLOR,
+                        textColor: BUTTON_TRANSITION_COLOR
+                    });
+                }
+            });
+
+            var soundPosition = Entities.getEntityProperties(courtyard, 'position').position;
+            _this.playSound(TRANSITION_SOUND, TRANSITION_VOLUME, soundPosition, false, true);
+            
+            if (!roundtableInPlace && roundtableVelocity) {
+                Entities.editEntity(roundtable, {
+                    velocity: { x: 0, y: roundtableVelocity, z: 0 }
+                });
+            }
+
+            if (!courtyardInPlace && courtyardVelocity) {
+                Entities.editEntity(courtyard, {
+                    velocity: { x: 0, y: courtyardVelocity, z: 0 }
+                });
+            }
+
+            if (checkInterval) {
+                Script.clearInterval(checkInterval);
+                checkInterval = null;
+            }
+
+            checkInterval = Script.setInterval(function() {
+                _this.checkEntityPositions(buttonID, targetCourtyardPosition, targetRoundtablePosition);
+            }, CHECK_INTERVAL_MS);
+        },
+
+        /* Check each moveable entity and if it is still moving and close enough to its target position, slam it and clear 
+        its velocity. If both entities are in place, play the done sound and set the button color to active and stop the 
+        position checking interval.
+        */
+        checkEntityPositions: function(buttonID, targetCourtyardPosition, targetRoundtablePosition) {
+            if (!courtyardInPlace) {
+                var courtyardCurrentPosition = Entities.getEntityProperties(courtyard, 'position').position;
+                var courtyardDistanceToTarget = courtyardCurrentPosition.y.toFixed(FIXED_DECIMAL_PLACES) - 
```

Comment:
- Why are you using `toFixed` here?
- Why not use `Vec3.distance()` here?
- This isn't a distance, since it could be negative. Did you intend for this to be a distance? If so, you'll need to wrap this in `Math.abs()` (but only if you're not going to use `Vec3.distance()`)

Reply:
I covered this above. It will be 'displacement' along the y axis instead of 'distance'. The entity only has animated movement along one axis so I only check that axis.  I used toFixed in order to round the numbers for comparison as the final value of the entity's position on the y axis, when rounded, should be equal to the the rounded value of the target position on the y axis. This could be changed to use an epsilon value to get a comparison but I feel it accomplishes the same goal.